src/Application.cpp

```diff
@@ -0,0 +1,230 @@
+//Copyright (c) 2018 Ultimaker B.V.
+//CuraEngine is released under the terms of the AGPLv3 or higher.
+
+#ifdef _OPENMP
+    #include <omp.h> // omp_get_num_threads
+#endif // _OPENMP
+#include <string>
+#include "Application.h"
+#include "FffProcessor.h"
+#include "communication/ArcusCommunication.h" //To connect via Arcus to the front-end.
+#include "communication/CommandLine.h" //To use the command line to slice stuff.
+#include "utils/logoutput.h"
+
+namespace cura
+{
+
+Application::Application()
+: communication(nullptr)
+, current_slice(0)
+{
+}
+
+Application::~Application()
+{
+    delete communication;
+}
+
+Application& Application::getInstance()
+{
+    static Application instance; //Constructs using the default constructor.
+    return instance;
+}
+
+#ifdef ARCUS
+void Application::connect()
+{
+    std::string ip = "127.0.0.1";
+    int port = 49674;
+
+    //Parse port number from IP address.
+    std::string ip_port(argv[2]);
+    if (ip_port.find(':') != std::string::npos)
```

Comment:
`ip_port.find(':') `  is used here thrice. Please factor it out into a separate variable. 

Or does this function change the underlying `ip_port` ?!

Reply:
It was like this already before, but it makes sense to me. Done in https://github.com/Ultimaker/CuraEngine/commit/d02eb0ca2ac4f9c50c7c8346fbdb86dcb1950a9e